ios/ImagePickerManager.m

```diff
@@ -292,10 +314,114 @@ + (UIImage*)getUIImageFromInfo:(NSDictionary*)info {
+            }];
+        });
+        return;
+    }
+
+    for (PHPickerResult *result in results) {
+        NSItemProvider *provider = result.itemProvider;
+
+        if ([provider canLoadObjectOfClass:[UIImage class]]) {
+            [provider loadObjectOfClass:[UIImage class]
+                      completionHandler:^(__kindof id<NSItemProviderReading> _Nullable object, NSError * _Nullable error) {
+                UIImage *image = object;
+                NSDictionary *response = [self makeResponseFromImage:image];
+
+                dispatch_async(dispatch_get_main_queue(), ^{
+                    [picker dismissViewControllerAnimated:YES completion:^{
+                        self.callback(@[response]);
+                    }];
+                });
+            }];
+        }
+        else if ([provider hasItemConformingToTypeIdentifier:@"public.movie"]) {
+            [provider loadFileRepresentationForTypeIdentifier:@"public.movie"
+                                            completionHandler:^(NSURL * _Nullable url, NSError * _Nullable error) {
+                NSDictionary *response = [self makeResponseFromURL:url];
+
+                dispatch_async(dispatch_get_main_queue(), ^{
+                    [picker dismissViewControllerAnimated:YES completion:^{
+                        self.callback(@[response]);
+                    }];
+                });
+            }];
+        }
+    }
+}
+
+- (NSDictionary *)makeResponseFromImage:(UIImage *)image {
+    NSMutableDictionary *response = [[NSMutableDictionary alloc] init];
+
+    NSData *data = UIImagePNGRepresentation(image);
+    NSString *fileType = [ImagePickerUtils getFileType:data];
+    if (![fileType isEqualToString:@"gif"]) {
+        image = [ImagePickerUtils resizeImage:image
+                                     maxWidth:[self.options[@"maxWidth"] floatValue]
+                                    maxHeight:[self.options[@"maxHeight"] floatValue]];
+    }
+
+    response[@"type"] = [@"image/" stringByAppendingString:fileType];
+
+    NSString *fileName = [self getImageFileName:fileType];
+    NSString *path = [[NSTemporaryDirectory() stringByStandardizingPath] stringByAppendingPathComponent:fileName];
+    [data writeToFile:path atomically:YES];
+
+    if ([self.options[@"includeBase64"] boolValue]) {
+        response[@"base64"] = [data base64EncodedStringWithOptions:0];
+    }
+
+    NSURL *fileURL = [NSURL fileURLWithPath:path];
+    response[@"uri"] = [fileURL absoluteString];
+
+    NSNumber *fileSizeValue = nil;
+    NSError *fileSizeError = nil;
+    [fileURL getResourceValue:&fileSizeValue forKey:NSURLFileSizeKey error:&fileSizeError];
+    if (fileSizeValue){
+        response[@"fileSize"] = fileSizeValue;
+    }
+
+    response[@"fileName"] = fileName;
+    response[@"width"] = @(image.size.width);
+    response[@"height"] = @(image.size.height);
+
+    return [response copy];
+}
+
+- (NSDictionary *)makeResponseFromURL:(NSURL *)url {
```

Comment:
makeResponseFromURL and onVideoObtained should be merged into a common function.